api/s2n.h

```diff
@@ -76,6 +85,7 @@ extern int s2n_config_set_status_request_type(struct s2n_config *config, s2n_sta
 typedef enum { S2N_CT_SUPPORT_NONE = 0, S2N_CT_SUPPORT_REQUEST = 1 } s2n_ct_support_level;
 extern int s2n_config_set_ct_support_level(struct s2n_config *config, s2n_ct_support_level level);
 extern int s2n_config_set_extension_data(struct s2n_config *config, s2n_tls_extension_type type, const uint8_t *data, uint32_t length);
+extern int s2n_config_set_max_fragment_length(struct s2n_config *config, uint8_t mfl_code);
```

Comment:
**s2n_config_set_max_fragment_length** allows the caller to set a TLS Max
Fragment Length extenstion that will be used to fragment outgoing messages.
s2n currently do not reject over-the-max incoming fragments. The TLS negotiated
max fragment length overrides the preference set by the
**s2n_connection_prefer_throughput** and **s2n_connection_prefer_low_latency**.

Additionally, I took out "Connections prefer low latency by default." comment in USAGE-GUIDE as this is not true based on what I see in the code.